cwms-data-api/src/main/java/cwms/cda/data/dto/basin/Basin.java

```diff
@@ -0,0 +1,138 @@
+package cwms.cda.data.dto.basin;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.data.dto.LocationIdentifier;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV1;
+
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class, aliases = {Formats.DEFAULT, Formats.JSON})
+@JsonDeserialize(builder = Basin.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class Basin implements CwmsDTOBase {
+    private final LocationIdentifier basinId;
+    private final Double sortOrder;
+    private final Double totalDrainageArea;
+    private final Double contributingDrainageArea;
+    private final LocationIdentifier parentBasinId;
+    private final String areaUnit;
+    private final LocationIdentifier primaryStreamId;
+
+    private Basin(Builder builder)
```

Comment:
Review source code formatter and make sure you are using the correct one for this repo.

Reply:
Matched to Google Checkstyle except indentation

Reply:
We're close to the google checkstyle but not quite. see `config/checkstyle/checkstyle.xml`

(We don't currently recommend running the checkstyle task, or at the least ignore the errors on work that isn't yours.)

Reply:
Imported the checkstyle.xml and used it to compare, no more changes to be made